### PR TITLE
[PUBDEV-7793] implement deserialization of monotone constraints

### PR DESCRIPTION
**Pull request by honzasterba**
…m in MOJO

### CODE REVIEW

**Review by Pscheidl — APPROVED** [review]

File: h2o-genmodel/src/main/java/hex/genmodel/attributes/parameters/KeyValue.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package hex.genmodel.attributes.parameters;
+
+public class KeyValue {
```
Comment (mn-mikke) [inline h2o-genmodel/src/main/java/hex/genmodel/attributes/parameters/KeyValue.java:3]: Can't we use `water.Iced.KeyValue` instead or is there any limitation?

File: h2o-test-support/src/main/java/water/TestUtil.java [hunk 2]
```diff
@@ -1371,4 +1374,21 @@ public Vec createRandomBinaryWeightsVec(final long len, final int randomSeed) {
     return weightsVec;
   }
 
+  public static GenModel toMojo(Model model, String testName, boolean readModelMetaData) {
+    final String filename = testName + ".zip";
+    StreamingSchema ss = new StreamingSchema(model.getMojo(), filename);
+    try {
+      FileOutputStream os = new FileOutputStream(ss.getFilename());
+      ss.getStreamWriter().writeTo(os);
```
Comment (mn-mikke) [inline h2o-test-support/src/main/java/water/TestUtil.java:1382]: if this line throws an exception, does `os` get closed?

File: h2o-genmodel/src/main/java/hex/genmodel/attributes/parameters/KeyValue.java (hunk 1 above)
Comment (Pscheidl) [reply h2o-genmodel/src/main/java/hex/genmodel/attributes/parameters/KeyValue.java:3]: This is MOJO

File: h2o-test-support/src/main/java/water/TestUtil.java (hunk 2 above)
Comment (honzasterba) [reply h2o-test-support/src/main/java/water/TestUtil.java:1382]: good catch, improved

**Review by mn-mikke — APPROVED** [review]
LGTM

**Review by Pscheidl — APPROVED** [review]

**Pull request merged**